### PR TITLE
feat(next): add next-auth generator

### PR DESCRIPTION
**Pull request by cloudratha**
Next:
Added NextAuth generator

Common/Core:
Refactored formatFilesWithEslint to use lint executor

Common/E2E:
Fixed an issue where publishable libraries referenced incorrect dependencies

Workspace:
Renamed install generator to init to match Next

Create:
Refactored based on workspace rename

### CODE REVIEW

File: packages/common/e2e/src/executors/e2e/executor.ts [hunk 1]
```diff
@@ -107,33 +107,100 @@ export default async function runEnd2EndExecutor(
...
+                                context,
+                            ),
+                            noBuild: true,
+                            packageVersion: version,
+                        },
+                        context,
+                    );
+            },
+        );
+        // const publishPromises = publishableLibraries.map(library => {
```
Comment (nandorszentpeteri) [inline packages/common/e2e/src/executors/e2e/executor.ts:177]: Did you leave this section on purpose?

File: packages/next/src/generators/next-auth/files/pages/api/[...nextauth].ts__template__ [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import NextAuth from 'next-auth';
+
+const nextAuth = NextAuth({
+    providers: []
+})
+
+export default nextAuth;
```
Comment (nandorszentpeteri) [inline packages/next/src/generators/next-auth/files/pages/api/[...nextauth].ts__template__:7]: You missed the empty line here ;)

File: packages/next/src/generators/next-auth/generator.spec.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        const appTs = appTree.read('next-app/pages/_app.tsx');
+
+        expect(appTs.toString()).toContain(
+            '<SessionProvider session={session}>',
+        );
+        expect(appTs.toString()).toContain('</SessionProvider>');
+    });
+
+    it('should configure app if there are already wrapping react providers', async () => {
+        const app = `function SomeProvider({ children }: PropsWithChildren) {
```
Comment (nandorszentpeteri) [inline packages/next/src/generators/next-auth/generator.spec.ts:47]: Can we somehow improve this formatting?

File: packages/next/src/generators/next-auth/generator.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import { tsMorphTree, formatFilesWithEslint } from '@ensono-stacks/core';
+import {
+    generateFiles,
+    joinPathFragments,
+    Tree,
+    getProjects,
+    logger,
+    formatFiles,
+} from '@nrwl/devkit';
+import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
```
Comment (nandorszentpeteri) [inline packages/next/src/generators/next-auth/generator.ts:10]: should we move this to 'core'?

**Review by nandorszentpeteri — APPROVED** [review]

File: packages/next/src/generators/next-auth/generator.ts (hunk 4 above)
Comment (cloudratha) [reply packages/next/src/generators/next-auth/generator.ts:10]: No I dont think so, unless we maintain our own version. not all packages currently depend on core.

**Pull request merged**